extension/src/time_vector.rs

```diff
@@ -368,6 +368,95 @@ CREATE AGGREGATE rollup(\n\
     ],
 );
 
+#[pg_schema]
+pub mod toolkit_experimental {
+    use super::*;
+
+    // Only making this available through the arrow operator right now, as the semantics are cleaner that way
+    pub fn asof_join<'a, 'b>(
+        from: Timevector_TSTZ_F64<'a>,
+        into: Timevector_TSTZ_F64<'b>,
+    ) -> TableIterator<
+        'a,
+        (
+            name!(value1, Option<f64>),
+            name!(value2, f64),
+            name!(time, crate::raw::TimestampTz),
+        ),
+    > {
+        assert!(
+            from.num_points > 0 && into.num_points > 0,
+            "both timevectors must be populated for an asof join"
```

Comment:
Can we test these two errors?

Reply:
Added...it's actually surprisingly difficult to create an empty timevector.